amundsen_application/api/metadata/v0.py

```diff
@@ -377,6 +377,75 @@ def get_tags() -> Response:
         return make_response(payload, HTTPStatus.INTERNAL_SERVER_ERROR)
 
 
+def _update_metadata_service(table_key: str, method: str, tag: str) -> int:
+    table_endpoint = _get_table_endpoint()
+    url = f'{table_endpoint}/{table_key}/tag/{tag}'
+    response = request_metadata(url=url, method=method)
+    status_code = response.status_code
+    if status_code != HTTPStatus.OK:
+        LOGGER.info(f"Fail to update tag in metadataservice, http status code: {status_code}")
+        LOGGER.debug(response.text)
+    return status_code
+
+
+def _update_search_service(table_key: str, method: str, tag: str) -> int:
+    searchservice_base = app.config['SEARCHSERVICE_BASE']
+    searchservice_get_table_url = f'{searchservice_base}/search_table'
+
+    # searchservice currently doesn't allow colon or / inside filters, thus can't get item based on key
+    # table key e.g: 'delta://gold.prod/workloads_sku_agg'
+    database = table_key[:table_key.find('://')]
+    schema = table_key[table_key.find('.') + 1:table_key.rfind('/')]
+    table = table_key[table_key.rfind('/') + 1:]
+    cluster = table_key[table_key.find('://') + 3:table_key.find('.')]
+
+    request_param_map = {
+        "search_request":
+            {
+                "type": "AND",
+                "filters":
+                    {
+                        "database": database,
+                        "schema": schema,
+                        "table": table,
+                        "cluster": cluster
+                    }
+            },
+        "query_term": ""
+    }
+
+    get_table_response = request_search(url=searchservice_get_table_url, method='POST', json=request_param_map)
+    get_status_code = get_table_response.status_code
+    if get_status_code != HTTPStatus.OK:
+        LOGGER.info(f"Fail to get table info from serviceservice, http status code: {get_status_code}")
+        LOGGER.debug(get_table_response.text)
+        return get_status_code
+
+    raw_data_map = json.loads(get_table_response.text)
+    # key is unique, thus (database, cluster, schema, table) should uniquely identify the table
+    table = raw_data_map['results'][0]
```

Comment:
does the result have only single element?

Reply:
Yes, if everything works correctly, we can add checks here